### PR TITLE
RDCC-5815-Common data api - Test coverage with BDD Approach

### PR DESCRIPTION
**Pull request by manukundloo-hmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-5900

### Change description ###
RDCC-5900-Common data api Controller - Test coverage with BDD Approach

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CrdApiControllerTest.java [hunk 1]
```diff
@@ -1,101 +1,116 @@
...
+    private GlobalExceptionHandler globalExceptionHandler;
+
+    @Autowired
+    private MockMvc mockMvc;
 
     @Test
-    void testWithValidCategoryId_ShouldReturnStatusCode200() {
-        CategoryRequest request = buildCategoryRequest(null, null, null,
-                                                       null, null, "N");
+    void testShouldReturn200WhenPassingRightCategoryId() throws Exception {
```
[the remark this answers is not in the document]
Comment (manukundloo-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CrdApiControllerTest.java:56]: thank you for the review comment add the displayname annotation before the test case

File: src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiControllerTest.java [hunk 2]
```diff
@@ -1,83 +1,180 @@
...
-        verify(caseFlagService, times(1))
-            .retrieveCaseFlagByServiceId("XXXX", "PARTY", "N");
-    }
 
     @Test
-    void testGetCaseFlag_ByInvalidFlag_Returns400() {
-        assertThrows(InvalidRequestException.class, () ->
-            caseFlagApiController.retrieveCaseFlagsByServiceId(
-                "XXXX", "Hello", ""));
+    void should_return_200_with_all_positive_scenario_flags() throws Exception {
```
[the remark this answers is not in the document]
Comment (manukundloo-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiControllerTest.java:57]: naming convention followed for all test cases

[the remark this answers is not in the document]
Comment (manukundloo-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiControllerTest.java:57]: display name annotation addd for all test cases

File: src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiControllerTest.java [hunk 3]
```diff
@@ -1,83 +1,180 @@
...
+                    .queryParam("flag-type", "PARTY")
+                    .queryParam("welsh-required", "Y")
+                    .accept(MediaType.APPLICATION_JSON_VALUE))
+            .andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.flags", hasSize(1)))
+
+            .andExpect(jsonPath("$.flags[0].FlagDetails", hasSize(1)))
+            .andExpect(jsonPath("$.flags[0].FlagDetails[0].name", is(name)))
+            .andExpect(jsonPath("$.flags[0].FlagDetails[0].flagCode", is(flagCode)))
```
[the remark this answers is not in the document]
Comment (manukundloo-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiControllerTest.java:79]: Added assertion for other fields

File: src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiControllerTest.java [hunk 4]
```diff
@@ -1,83 +1,180 @@
...
+
+            .andExpect(jsonPath("$.flags[0].FlagDetails[0].childFlags", hasSize(1)))
+            .andExpect(jsonPath("$.flags[0].FlagDetails[0].childFlags[0].name", is(name)))
+            .andExpect(jsonPath("$.flags[0].FlagDetails[0].childFlags[0].flagCode", is(flagCode)));
+
```
[the remark this answers is not in the document]
Comment (manukundloo-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CaseFlagApiControllerTest.java:84]: Added assertion for child flag details

File: src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CrdApiControllerTest.java [hunk 5]
```diff
@@ -1,101 +1,116 @@
...
+        //when
+        given(crdService.retrieveListOfValuesByCategory(any(CategoryRequest.class))).willReturn(categoryList);
 
-        when(crdService.retrieveListOfValuesByCategory(any(CategoryRequest.class))).thenReturn(categoryList);
+        //then
+        mockMvc.perform(get("/refdata/commondata//lov/categories/{categoryId}", "HearingChannel")
+                            .accept(MediaType.APPLICATION_JSON_VALUE))
+            .andDo(print())
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.list_of_values", hasSize(1)));
```
[the remark this answers is not in the document]
Comment (manukundloo-hmcts) [reply src/test/java/uk/gov/hmcts/reform/cdapi/controllers/CrdApiControllerTest.java:68]: Added assertion for content and child nodes

**Review by rajkatla-hmcts — APPROVED** [review]

**Review by sahitya-desireddy — APPROVED** [review]

**Pull request merged**